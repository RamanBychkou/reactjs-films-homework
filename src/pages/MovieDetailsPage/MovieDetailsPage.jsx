import React from 'react';
import Header from '../../components/Header';
import MainfilmTitle from '../../components/MainFilmTitle';
import MainFilmInfo from '../../components/MainFilmInfo';

import style from './MovieDetailsPage.scss';

const MovieDetailsPage = () => (
  <section className={style.movieDetailsPageContainer}>
    <Header />
    <div className={style.mainfilmInfo}>
      <div className={style.container}>
        <MainfilmTitle />
        <MainFilmInfo />
      </div>
    </div>
  </section>
);

export default MovieDetailsPage;
