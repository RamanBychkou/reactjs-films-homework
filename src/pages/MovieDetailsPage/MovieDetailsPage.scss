.movieDetailsPageContainer {
  background-image: url(./img/mainFilmBG.png);
  background-repeat: no-repeat;
  background-size: cover;
  background-position: 0 0;
  display: flex;
  flex-flow: row wrap;
  align-content: space-between;
  min-height: 34.2rem;
}

.mainfilmInfoBlock {
  display: flex;
  flex-flow: row nowrap;
  justify-content: space-between;
}

.mainfilmInfo {
  padding: 2rem 0.8rem 1.4rem 4.2rem;
  width: 100%;
}

.container {
  max-width: 82rem;
  width: 100%;
  display: flex;
  justify-content: space-between;
  flex-flow: row wrap;
}

@media screen and (min-width: 82rem) {
  .container {
    margin: 0 auto;
  }
}

@media screen and (max-width: 46.8rem) {
  .container {
    justify-content: center;
  }
}
