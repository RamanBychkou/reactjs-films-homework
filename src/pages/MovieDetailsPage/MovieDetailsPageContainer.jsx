import React from 'react';
import Header from '../../components/Header';
import MainfilmTitle from '../../components/MainFilmTitle';
import MainFilmInfo from '../../components/MainFilmInfo';

import style from './MovieDetailsPageContainer.scss';

const MovieDetailsPage = () => (
  <section className={style.movieDetailsPageContainer}>
    <Header />
    <div className={style.mainfilmInfo}>
      <MainfilmTitle />
      <MainFilmInfo />
    </div>
  </section>
);

export default MovieDetailsPage;
