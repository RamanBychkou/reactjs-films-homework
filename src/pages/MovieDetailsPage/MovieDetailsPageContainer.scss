.movieDetailsPageContainer {
  background-image: url(./img/mainFilmBG.png);
  background-repeat: no-repeat;
  background-size: cover;
  background-position: 0 0;
  display: flex;
  flex-flow: row wrap;
  align-content: space-between;
  min-height: 34.2rem;
}

.mainfilmInfoBlock {
  display: flex;
  flex-flow: row nowrap;
  justify-content: space-between;
}

.mainfilmInfo {
  display: flex;
  justify-content: space-between;
  flex-flow: row wrap;
  width: 100%;
  padding: 2rem 0.8rem 1.4rem 0;
}

@media screen and (max-width: 46.8rem) {
  .mainfilmInfo {
    justify-content: center;
  }
}
