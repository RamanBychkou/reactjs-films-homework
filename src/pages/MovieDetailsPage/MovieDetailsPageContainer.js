import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import MovieDetailsPage from './MovieDetailsPage';
import requestsFilms from '../../utils/requests';
import clearError from '../../modules/Error/errorAction';
import * as fromFetchGenres from '../../modules/fetchGenres/fetchGenresAction';
import * as fromClearCurrentMovie from '../../modules/root/clearCurrentMovieAction';

const mapStateToProps = state => state;

export const mapStateToDispatch = dispatch => ({
  fetchVideo: requestsFilms.fetchVideo.bind(this, dispatch),
  clearError: clearError.bind(dispatch),
  getMainMovieDetails: requestsFilms.getMainMovieDetails.bind(this, dispatch),
  fetchSearchResults: requestsFilms.fetchSearchResults.bind(this, dispatch),
  fetchListMovies: requestsFilms.fetchListMovies.bind(this, dispatch),
  fetchGenres: fromFetchGenres.default.bind(null, dispatch),
  clearCurrentMovie: fromClearCurrentMovie.default.bind(null, dispatch),
});
export default withRouter(
  connect(
    mapStateToProps,
    mapStateToDispatch,
  )(MovieDetailsPage),
);
