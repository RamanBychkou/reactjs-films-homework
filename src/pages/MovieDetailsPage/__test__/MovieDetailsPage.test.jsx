import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import MovieDetailsPage from '../index';


test('MovieDetailsPage renders correctly', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<MovieDetailsPage />);
  const result = renderer.render(<MovieDetailsPage />);
  expect(result).toMatchSnapshot();
});
