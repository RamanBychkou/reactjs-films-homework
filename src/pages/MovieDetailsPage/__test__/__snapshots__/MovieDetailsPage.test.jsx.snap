// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetailsPage renders correctly 1`] = `
<section
  className="movieDetailsPageContainer"
>
  <Header />
  <div
    className="mainfilmInfo"
  >
    <MainFilmTitle />
    <MainFilmInfo />
  </div>
</section>
`;
